src/hotspot/cpu/aarch64/gc/shenandoah/shenandoahBarrierSetAssembler_aarch64.hpp

```diff
@@ -59,7 +60,7 @@ class ShenandoahBarrierSetAssembler: public BarrierSetAssembler {
   void resolve_forward_pointer_not_null(MacroAssembler* masm, Register dst, Register tmp = noreg);
   void load_reference_barrier(MacroAssembler* masm, Register dst, Address load_addr);
   void load_reference_barrier_not_null(MacroAssembler* masm, Register dst, Address load_addr);
-  void load_reference_barrier_native(MacroAssembler* masm, Register dst, Address load_addr);
+  void load_reference_barrier_native(MacroAssembler* masm, Register dst, Address load_addr, bool native);
 
```

Comment:
I think it's clearer as it is. The motivation for this is that native references are always oops, while weak reference's referents can be oops or narrowOops. Which means that we need to call a different method for native-refs (the always-oops entry point).
The interesting differentiator is native vs. not-native, because one is always-oops, the other can be narrowOop. Weak vs not-weak is not as clear because there can also be weak/phantom native-refs.

Reply:
The weird part is the method name, load_reference_barrier_**native**(), then you have parameter says that it may not be a native load and the parameter does nothing to hint the possible type differential. How about may_narrow_oop?

Reply:
Ok I renamed the argument. It is indeed better to decouple the meanings there native <-> narrow/wide.